test/Feature/AuthSpec.hs

```diff
@@ -31,15 +31,36 @@ spec = beforeAll
     request methodGet "/authors_only" [auth] ""
       `shouldRespondWith` 200
 
+  it "works with tokens which have extra fields" $ do
+    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoicG9zdGdyZXN0X3Rlc3RfYXV0aG9yIiwiaWQiOiJqZG9lIiwia2V5MSI6InZhbHVlMSIsImtleTIiOiJ2YWx1ZTIiLCJrZXkzIjoidmFsdWUzIiwiYSI6MSwiYiI6MiwiYyI6M30.GfydCh-F4wnM379xs0n1zUgalwJIsb6YoBapCo8HlFk"
```

Comment:
Yay for all these tests!

What if you created these test jwt values by calling the encoder function on the plaintext json rather than embedding encoded jwt in the code? In this case

```haskell
let auth = tokenJWT secret
      [json| {
        "role": "postgrest_test_author",
        "id": "jdoe",
        "key1": "value1",
        "key2": "value2",
        "key3": "value3",
        "a": 1, "b": 2, "c": 3
      } |]
```